### DISCUSSION TITLE
[V0.1400.0.0] Pirate mission feedback and tweaks (pirate ship rework inside)

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#5853] [Unstable feedback]
Pirate missions are pretty fun and dangerous, but here are few things to consider:

**The amount of loot is too much (when sinking)**

The amount of loot in them is probably worth about 10k if not more. The armory alone is worth several thousand, the cabinets are all on autofill and also full of valuable items:
![image](https://user-images.githubusercontent.com/73229309/119135078-3e3f3300-ba3e-11eb-883b-5ba1bb3c4ff8.png)

**Make BOARDING more rewarding than SINKING**

You have two ways to approach pirate ships:
-Boarding (high risk, high reward)
-Sinking (low risk, low reward)
Since sinking basically turns the ship into a wreck, it's also "high reward"; you get to loot all the contents of the sub after it sinks to the sea floor. Pop a few explosive railgun shells in it from a distance and you're rich.

In short, you should get less loot from *sinking* the ship and using brute force, and more loot for *boarding* and keeping the sub functional
But how?

I've devised a **self-destruct mechanism** with a nuke on pirate Humpback: if the submarine is almost entirely flooded, a nuke will be detonated after 30 seconds. This means you can't safely loot the ship after sinking it, as it will be irradiated (radiation is still weak, but a deterrent). I left annotations and small tweaks to the top airlock, the circuit can be easily copied in other pirate ships

**Feature request:** Another solution would be "destroyable" containers. Cabinets with a condition factor, that you cannot open when destroyed. Firing on those containers would break the loot, so you'd need to be more careful when boarding.

**Balance:**
Coilguns do very little damage to divers; I was able to approach the top hatch of a humpback and rewire/crowbar it under coilgun fire, while keeping about 80% of my life. This has to do with damage resistance in general.
The armory is right next to the top airlock, so it needs to be id restricted or you can just rush the enemy grenades
AI has perfect railgun accuracy; you WILL get sniped while scooting outside, or even piloting small agile shuttles around the sub

The sub:
[HumpbackPirateTweaked.zip](https://github.com/Regalis11/Barotrauma/files/6535199/HumpbackPirateTweaked.zip)

Changelog:
-Rewired top hatch, so it can be opened with crowbar, and closes by itself again to allow boarding
-Self-destruction mechanism, with 60 sec countdown, alarm and wifi messaging
-Added id_pirate to all spawnpoints to allow for id-locking of doors and cabinets. Armory is id locked
-Slowed down the railgun rotation speed + added delay to counteract the AI's perfect accuracy
-Added a bottom docking port for boarding shuttles

**Comment by Udrakan** [comment]
**About loot:**
Pirate subs behaving differently with the explosives would be weird. I'd much rather have their loot hard nerfed. If they had such great amount of supplies, they wouldn't need to be pirates.

**About boarding:**
The ultimate mechanic for me would be having the ability to arrest the pirates for reward. That would highly incentivise boarding compared to sinking, but it would require a whole bunch of new features that likely wont make it to the game. 

**Looting in general**
Looting wrecks, pirate ships and abandoned outposts in Baro is 10% action, 90% inventory management, which makes me not excited for those features, which are very inviting in concept. I'd change the loot philosophy of these places completely

-end cabinet with 1-3 interesting items (alien ores, unique wreck weapons, oxygenite tank etc.)
-two or three cabinets containing 3-5 medium tier items (medical items, high tier ores, revolver, SMG, harpoon gun, fuel rod)

**Reply by ubertpendragon** [reply]
> **Looting in general**
> Looting wrecks, pirate ships and abandoned outposts in Baro is 10% action, 90% inventory management, which makes me not excited for those features, which are very inviting in concept. I'd change the loot philosophy of these places completely

Looting should indeed be a minor part of Pirate/Outpost missions, and obviously the more loot, the more time you spend moving things around in crates back and forth to your sub. 

The event manager and the rad wall to a lesser extent are supposed to pressure you into not spending too much time looting but they're mostly irrelevant in the early, easier biomes

Wrecks having less action is fine, but the "action" was supposed to be replace with "exploration", unfortunately:
-We don't have enough wrecks
-Wreck layouts are always the same

I've tried using explosives and ice pillars to randomize wrecks; and it works, but I wish we had a system ala Thalamus to add random debris and disable loot cabinets

**Comment by Toybasher** [comment]
First, I want the pirate AI overhauled somehow. IIRC they "cheat" by using X-ray Vision and shoot at your crew, not your sub. (I get why though, considering the AI doesn't recognize the sub as hostile, so this is a workaround.) I want to see this eventually changed, (AI targets the sub) because in my experience the pirates are often sharpshooters and love to knock people off the guns by aiming for the gunner through the hull. "Detection" could also be improved, so the AI has to spot you on sonar instead of seemingly having a spidey-sense. 

I'd love to see the pirates having different AI "personalities" affecting how they react and fight against you. Some pirate crews remain passive on sonar until they spot you, others start active from the beginning. (Emitting orange waves similar to an endworm, indicating they are active on their sonar.) Some aggressively attempt to dock with you and board you, some send out divers with plasma cutters to try and board, some act like the current pirates and mostly stay on their onboard weapons, some try to sink you then board during the chaos, when you dock with pirates, some attempt to undock like currently, but others see it as a challenge and gear up and rush you, etc. 

Overall I agree with pirates battles involving sinking eachother too often. I want to see pirates who aggressively try to board _you_ for a change.

EDIT: I am curious how your humpback nuke works. Does it just trigger a reactor meltdown when flooded? Your initial post claims 30 seconds but in the changelog it's mentioned to be 60 seconds.

**Reply by ubertpendragon** [reply]
> EDIT: I am curious how your humpback nuke works. Does it just trigger a reactor meltdown when flooded? Your initial post claims 30 seconds but in the changelog it's mentioned to be 60 seconds.

It works by checking the water% in every room and does an average; above a certain % the sub is considered "sunk" and it triggers the countdown

Overall it's a convoluted solution to the sink > board problem. 
The game probably needs a mission bonus system instead
